src/worker/jobs/downloads/clean_log_dedupe_table.rs

```diff
@@ -0,0 +1,109 @@
+use crate::schema::processed_log_files;
+use crate::worker::Environment;
+use anyhow::anyhow;
+use crates_io_worker::BackgroundJob;
+use diesel::prelude::*;
+use std::sync::Arc;
+
+/// This job is responsible for cleaning up old entries in the
+/// `processed_log_files` table.
+///
+/// Rows older than one week will be deleted.
+#[derive(Serialize, Deserialize)]
+pub struct CleanLogDedupeTable;
+
+impl BackgroundJob for CleanLogDedupeTable {
+    const JOB_NAME: &'static str = "clean_log_dedupe_table";
+    const QUEUE: &'static str = "downloads";
+
+    type Context = Arc<Environment>;
+
+    async fn run(&self, env: Self::Context) -> anyhow::Result<()> {
+        let conn = env.deadpool.get().await?;
+        conn.interact(run)
+            .await
+            .map_err(|err| anyhow!(err.to_string()))??;
+
+        Ok(())
+    }
+}
+
+fn run(conn: &mut PgConnection) -> QueryResult<()> {
+    let cut_off_date = chrono::Utc::now() - chrono::Duration::try_weeks(1).unwrap();
```

Comment:
`try_weeks` :grimacing:

Reply:
don't get me started........

Reply:
We can also achieve this using diesel:
```rust
use diesel::dsl::{now, IntervalDsl};

let cut_off_date = now.into_sql::<diesel::sql_types::Timestamptz>() - 1.weeks();
```
similar to how it's done in the https://github.com/rust-lang/crates.io/blob/a114476740872411f07e2be746a63fbfe391e8aa/src/controllers/krate/publish.rs#L429-L437